src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -31,6 +31,21 @@ protected CompareExchangeCommandBase(string key, long index, JsonOperationContex
 
         public abstract (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index);
 
+        public unsafe bool Validate(TransactionOperationContext context, Table items, long index, out long currentIndex)
+        {
+            var dbKey = Key.ToLowerInvariant();
+            currentIndex = -1;
+            using (Slice.From(context.Allocator, dbKey, out Slice keySlice))
+            {
+                if (items.ReadByKey(keySlice, out var reader))
```

Comment:
What if the value doesn't not exists ?

Reply:
will return `true` and currentIndex `-1`